### DISCUSSION TITLE
关于词汇选定后上屏前的变换

### DISCUSSION THREAD

**Discussion by alephpi** [rime/home#1184] [Q&A] [answered]
您好，我现在正在设计一款英文输入方案，雏形大定，现在的有这么几个feature暂时不知如何实现：

1. 空格上屏前给每个单词自动添加空格
2. 按某个功能键（如shift）将单词自动全部大写上屏

关于第一点，我想类比符号自动上屏，比如输入逗号后逗号自动添加到单词末尾并上屏。但是输入空格似乎不能识别为标点符号。我尝试了
```
key_binder/bindings:
  - {when: always, accept: space, send_sequence: "{space}{space}" }
```
不起作用。我又注意到负责空格上屏逻辑的express_editor，但我没发现如何定制该组件。

关于第二点，还没有头绪，我看了wiki中似乎没有对单词选定后上屏前这个时机作处理的逻辑。
不知道以上特性能否通过配置yaml文件实现。抑或需要自己写插件？

在下Rime新手，请诸位大佬不吝赐教。

**Comment by lotem** [comment, marked as answer]
1. 逗号，那个不是自动添加到单词末尾，而是逗号配置为 `commit` 即上屏动作。Rime先对输出串按输入内容类型分段，逗号触发上屏动作时，有完成转换的一段拼音和一段符号，一起上屏——按下逗号键的一瞬已经默认选定了那段拼音对应的候选字。那么空格作为字符上屏的方法是设置 `punctuator/use_space: true` 并在 `punctuator/half_shape` 里把空格定义为上屏的标点。还要保证 `punctuator` 排列顺序在把空格当作候选字确认键的 `selector` 之前。
2. 同时上屏不支持的，因为这时候也没法指明上屏哪个候选不是。大小写转换比较容易，可以写一个 `Filter`，用 librime 插件、lua 脚本、opencc 都可以，负责在某个开关打开时做大小写转换；快捷键绑定到这个开关上。

**Reply by alephpi** [reply]
第一项功能实现了，非常感谢！有时间学学第二项怎么做。
此外有一个细节，就是这种情况下`speller/delimiter`中不能含有空格（我在这里绊了好久）
